clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -753,22 +754,7 @@ public static ByteBuffer ensureCapacity(ByteBuffer existingBuffer, int newLength
      * @return An entry
      */
     public static <K, V> Map.Entry<K, V> mkEntry(final K k, final V v) {
-        return new Map.Entry<K, V>() {
-            @Override
-            public K getKey() {
-                return k;
-            }
-
-            @Override
-            public V getValue() {
-                return v;
-            }
-
-            @Override
-            public V setValue(final V value) {
-                throw new UnsupportedOperationException();
-            }
-        };
+        return new AbstractMap.SimpleEntry<>(k, v);
```

Comment:
The previous implementation did not implement `equals`, which caused one test to fail.